simplify code for reading tags

funniray opened the pull request:
also:
- Transformed seekHead object into a new object to make reading tags easier
- Throws error if file doesn't have a seekHead

funniray committed:
deduplicate code for reading tags

ThaUnknown commented on index.js:
```diff
@@ -106,180 +110,43 @@ export class Metadata {
-  const Cues = await metadata.getCues()
-  const Attachments = await metadata.getAttachments()
-  const Tags = await metadata.getTags()
+  const Info = await metadata.readSeekedTag("Info")
```
single quotes not double

ThaUnknown commented on index.js:
```diff
@@ -53,37 +53,41 @@ const readChildren = (parent = {}, start = 0) => {
-    this.cues = []
-    this.attachments = []
-    this.tags = []
+    this.Info = []
```
no.... keep them lowercase

ThaUnknown commented on index.js:
```diff
@@ -53,37 +53,41 @@ const readChildren = (parent = {}, start = 0) => {
+
+  const transformedHead = {}
+
+  seekHead.Children.forEach(child => {
```
since you're here, use for const of

ThaUnknown commented on index.js:
```diff
@@ -106,180 +110,43 @@ export class Metadata {
-        this.tags = readChildren(tags, this.segmentStart + seekHeadTags.Seek.SeekPosition)
-        return this.tags
+      const stream = this.blob.slice(this.segmentStart + this.seekHead[tag]).stream()
+      return readUntilTag(stream, EbmlTagId[tag], true).then((child) => {
```
since you're here revamp this to use await instead of .then

ThaUnknown commented on index.js:
```diff
@@ -106,180 +110,43 @@ export class Metadata {
-
-      if (!seekHeadTags) return Promise.resolve([])
+    if (this[tag].length === 0) {
+      if (!this.seekHead[tag]) return Promise.resolve([])
```
return []

funniray committed:
Minor changes to formatting

- Pass createStream to readSeekHead, as it's not part of the class
- Make info an object, not array

ThaUnknown commented on index.js:
```diff
@@ -50,25 +51,26 @@ const readChildren = (parent = {}, start = 0) => {
   if (transformedHead.SeekHead) {
-    const seekHeadStream = this.blob.slice(transformedHead.SeekHead + segmentStart).stream();
+    const seekHeadStream = createStream(transformedHead.SeekHead + segmentStart)
     return readSeekHead(seekHeadStream, segmentStart)
```
createStream is not passed here, also overall dont like the funcntion

funniray replied:
It's passed through from getSeekHead on line 114. Don't want to access this.blob from outside of the class. Could move `readSeekHead` inside of the class or just pass through the blob to the function

ThaUnknown replied:
its not passed to readSeekHead on L74

funniray replied:
From my understanding, you shouldn't have more than 1 recursive SeekHead, but might as well just in case

funniray committed:
make readSeekHead accept a blob

funniray committed:
keep tags from the first seek head

funniray commented:
I don't know how to implement this, but if the first non-crc32 element inside of the Segment isn't a SeekHead element, then we should assume that there's no SeekHead in the file and not try to read the entire file looking for a SeekHead
